### DISCUSSION TITLE
PopupBar is disappearing after hidesBottomBarWhenPushed

### DISCUSSION THREAD

**Discussion by STIFFMEISTER-UA** [LeoNatan/LNPopupController#531] [General] [answered]
**Describe the Bug**
The idea behind this bug is pretty simple. 
I have the following navigation stack:
UITabbarController -> UINavigationController -> UIViewController -> UIViewController.
Presenting PopupController from the UITabbarController. For the last UIViewController I must hide both tab bar and popup bar. 

Tab bar I'm hiding via `hidesBottomBarWhenPushed = true`, when pushing the vc.
Popup bar via `dismissPopupBar` in the tab bar controller from the viewWillAppear of the last vc. 
By this moment everything works as expected. The hitch here is that when I go back I need to show them both again. 
Tab bar is showing automatically. For showing popupBar I call ` presentPopupBar(withContentViewController: vc, animated: false)` in the tabBar from the viewWillDisappear of the vc. For a second the popupBar appears and then it vanishes leaving an empty space. Any ideas about what is going on here?

**To Reproduce**
UITabbarController -> UINavigationController -> UIViewController -> UIViewController.
Push the last vc with hidesBottomBarWhenPushed enabled and popupBar hidden. Try to present popupBar again on going back.

**Expected Behavior**
Popupbar should not disappear.

**Additional Context**
I have created a pet project to reproduce the issue.
[GitHub](https://github.com/STIFFMEISTER-UA/LnPopupExample)

**Comment by LeoNatan** [comment, marked as answer]
Hello,
Presenting/dismissing the popup bar during toolbar or tab bar hiding/showing transition is not supported. I suggest you present/dismiss the popup bar either in `viewDidAppear`/`viewDidDisappear`, or, preferably, using
```swift
override func viewWillDisappear(_ animated: Bool) {
	super.viewWillDisappear(animated)

	transitionCoordinator?.animate(alongsideTransition: nil, completion: { context in
		guard context.isCancelled == false else { return }
		
		self.tabBarController?.presentPopupBar(withContentViewController: UIViewController(), animated: true)
	})
}
```
